producers/pipsafety/main.go

```diff
@@ -0,0 +1,71 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+
+	v1 "api/proto/v1"
+
+	"github.com/thought-machine/dracon/producers"
+)
+
+type SafetyIssue struct {
+	Name              string
+	VersionConstraint string
+	CurrentVersion    string
+	Description       string
+}
+
+//read semi-unstructured safety json into struct
+func (i *SafetyIssue) UnmarshalJSON(data []byte) error {
+
+	var v []interface{}
+	if err := json.Unmarshal(data, &v); err != nil {
+		return err
+	}
+
+	i.Name, _ = v[0].(string)
+	i.VersionConstraint, _ = v[1].(string)
+	i.CurrentVersion = v[2].(string)
+	i.Description = v[3].(string)
+
+	return nil
+}
+
+func main() {
+	if err := producers.ParseFlags(); err != nil {
+		log.Fatal(err)
+	}
+	jsonBytes, err := ioutil.ReadFile(producers.InResults)
```

Comment:
We should be making use of `io.Reader` interface where we can. It meas the process doesn't have to load all the data in to memory at once, instead can just stream the data. To do this you'll need to use `json.Decoder` https://pkg.go.dev/encoding/json?tab=doc#example-Decoder